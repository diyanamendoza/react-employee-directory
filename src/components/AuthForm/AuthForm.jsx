import React, { useState } from 'react'
import styles from '../../App.css'

export default function AuthForm({handleAuth, heading}) {
    const [email, setEmail] = useState('')
    const [pw, setPw] = useState('')
    const [formError, setFormError] = useState('')

    const handleFormChange = (event) => {
        const { name, value } = event.target;
        if(name === 'email') setEmail(value)
        if(name === 'pw') setPw(value)
    }

    const handleSubmit = async (event) => {
        event.preventDefault()
        try {
            if (pw.length < 8) {
                setFormError('Password must be at least 8 characters long.')
                throw new Error}
           await handleAuth(email, pw)
        } catch (error) {
            console.log(error)
        }
    }

    return (
        <form className={styles.form} onSubmit={handleSubmit}>
        <fieldset className={styles.fieldset}>
          <legend>{heading}</legend>
          <section className={styles.formSection}>
            <label htmlFor="email">Email</label>
            <input
            required
              id="email"
              type="email"
              name="email"
              value={email}
              className={styles.input}
              onChange={handleFormChange}
            />
          </section>
          <section className={styles.formSection}>
            <label htmlFor="password">Password</label>
            <input
            required
              id="password"
              type="password"
              name="pw"
              value={pw}
              className={styles.input}
              onChange={handleFormChange}
            />
          </section>
          <button type='submit'>Submit</button>
          {/* <button type="submit" disabled={loading}>
            {loading ? 'Authenticating...' : heading}
          </button> */}
          {formError && <p>{formError}</p>}
        </fieldset>
      </form>
    )
}
