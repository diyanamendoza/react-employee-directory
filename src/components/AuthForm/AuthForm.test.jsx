import { render } from '@testing-library/react'
import AuthForm from './AuthForm'

it('should display a form for signing up/in', () => {
    const { container } = render(<AuthForm />);
    expect(container).toMatchSnapshot();
  });
  