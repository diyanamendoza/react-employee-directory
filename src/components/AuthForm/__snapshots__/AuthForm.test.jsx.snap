// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a form for signing up/in 1`] = `
<div>
  <form
    class="form"
  >
    <fieldset
      class="fieldset"
    >
      <legend />
      <section
        class="formSection"
      >
        <label
          for="email"
        >
          Email
        </label>
        <input
          class="input"
          id="email"
          name="email"
          required=""
          type="email"
          value=""
        />
      </section>
      <section
        class="formSection"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          class="input"
          id="password"
          name="pw"
          required=""
          type="password"
          value=""
        />
      </section>
      <button
        type="submit"
      >
        Submit
      </button>
      
    </fieldset>
  </form>
</div>
`;
